lib/project_types/theme/messages/messages.rb

```diff
@@ -32,6 +32,26 @@ module Messages
             APP
           },
         },
+        push: {
+          error: {
+            pushing_error: "Theme files couldn't be pushed to Shopify",
+            removing_error: "Theme files couldn't be removed from Shopify",
+          },
+          help: <<~HELP,
+            {{command:%s push}}: Replaces theme files on Shopify with those in your directory. If filenames are provided, only those files will be replaced; otherwise, the whole theme is replaced.
+              Usage: {{command:%s push}}
+              Options:
+                {{command:--remove}} Removes, rather than replaces, both local and Shopify copies of theme files. At least one file must be specified.
+                {{command:--allow-live}} Allows Shopify App CLI to replace a file on the currently live theme.
+                {{command:--nodelete}} Runs push without removing files from Shopify.
+          HELP
+          info: {
+            pushed: "Theme files were pushed from %s to Shopify",
+            removed: "Theme files were removed from %s and Shopify",
+          },
+          pushing: "Pushing theme files to Shopify",
+          removing: "Removing theme files from Shopify",
```

Comment:
Just wondering whether `delete` sounds scarier and more definitive than `remove` — and whether that's desirable. Curious what others think!

Reply:
I do think `delete` sounds scarier, but I'm not sure we want to scare developers at this point (removing the theme is scary enough IMHO) - the chicken part of me prefers `remove`.

Reply:
`delete` does sound scarier, but I prefer it - seems like a more permanent action, which seems to better convey what's happening.

Reply:
I think sometimes scarier is better — especially in the case of destructive actions! Prompting that moment of "yes, I'm _really_ sure" can be a feature, not a bug...

Reply:
That's actually especially true in this case since the command deletes the local version too! We might want to put some guardrails in the way of that, tbh

Reply:
Would these guardrails be a prompt for, "are you sure?"?

Reply:
That's one approach although it's not that common in CLI contexts. Another possibility would be to decouple the remote and local removal, so that the user has to actively choose:

```console
# delete the remote theme or file(s)
$ shopify push --remove remote
$ shopify push --remove remote /path/to/file

# delete the local theme or file(s)
$ shopify push --remove local /path/to/file

# delete both remote and local
$ shopify push --remove all
$ shopify push --remove all /path/to/file
```

I'm making this up as I go along, clearly, but given the high-risk nature of the activity I think it's worth thinking further about!

Reply:
I believe decoupling remote and local removal would be beyond the scope of this project, as it would go down to the Theme Kit level (`theme remove` does both local and remote). Would it make sense to add guardrails as a future consideration, but leave it as is for this PR?

Reply:
I think Graham's suggestions are great, but if they fall out of scope I'd go with a simple check just so we have some protection against people mistakenly dropping their entire theme.

Reply:
100% agree, this is scope creep via review thread. I think it's totally valid to simply mirror the existing Theme Kit functionality. If a simple double-check for safety is possible without a ton of trouble, that's a good compromise to me.